pkg/gatherers/clusterconfig/certificate_signing_requests.go

```diff
@@ -20,27 +20,33 @@ import (
 	"github.com/openshift/insights-operator/pkg/utils/anonymize"
 )
 
-// csrGatherLimit is the maximum number of crs that
-// will be listed in a single request to reduce memory usage.
-const csrGatherLimit = 5000
-
-// GatherCertificateSigningRequests collects anonymized CertificateSigningRequests.
-// Collects CSRs which werent Verified, or when Now < ValidBefore or Now > ValidAfter
+// GatherCertificateSigningRequests Collects anonymized CertificateSigningRequests which weren't Verified, or
+// when Now < ValidBefore or Now > ValidAfter
+//
+// ### API Reference
+// - https://github.com/kubernetes/client-go/blob/master/kubernetes/typed/certificates/v1beta1/certificatesigningrequest.go#L78
+// - https://docs.openshift.com/container-platform/4.3/rest_api/index.html#certificatesigningrequestlist-v1beta1certificates
+//
+// ### Sample data
+// - docs/insights-archive-sample/config/storage/openshift-storage/ocs-storagecluster-cephcluster.json
 //
-// The Kubernetes api:
+// ### Location in archive
+// | Version   | Path														|
+// | --------- | --------------------------------------------------------	|
+// | >= 4.5.0  | config/certificatesigningrequests 					        |
```

Comment:
This is slightly confusing. Was the gatherer available in 4.3.26 or not? This table says it wasn't but the "backports" section says it was. Which is true?

It took me some time to realize that you included the version here to cover file path changes, e.g. as in `pkg/gatherers/clusterconfig/config_maps.go`. The meaning is kind of clear there when there are multiple file paths with different versions. The simple greater than/smaller than ranges are probably good enough for that purpose because file path changes are unlikely to be backported.

@shzhou12 What do you think? Is this good enough or we want to try to find something better? 

I don't have a really better solution. Some other options could be:

* Use the table only when there have been some file path changes
* Use "all applicable" or similar if the file path hasn't changed
* Use the version of the farthest backport, i.e. `>=4.3.25` in this case

Reply:
Correct. I'm using it to track the path changes. I also questioned myself if the version should be the lowest backported version, but then I found out that we could have some weird cases. For example:

Some feature was released to `4.7.0` and backported to `4.6.14` and `4.5.22`. How should I mention it in the table? 'Cause if I use `>= 4.5.22` it means that `4.6.0` will also have the file, which isn't true, since for `4.6` it was only backported to `4.6.15+`.

I couldn't come up with a better solution for this. I'm not sure if `all applicable` or not using the table will help as well. I mean, it could help till a certain point, but it will not solve the problem, IMHO, until we include some path change to the gatherer and backport it then we are back to the problem.

Maybe by adding a `note` before the table saying that, if not specified in the table the backport uses the same path as the release version? 🤔

Reply:
When the table uses the "released version", there is a **gap**. The documentation doesn't say what the location is in backported versions. The reader is left to realize that the table states the "released version" and assume that backports used the same locations.

When we used the lowest backport, you are correct that there might be a **perceived contradiction** between the table and info about availability. The reader needs to know/realize that the table describes only file locations and not availability. Once they realize that, they don't have to make any further assumptions.

For me personally, the second case is easier to understand, but I acknowledge that this might be a matter of personal taste. :shrug:

Reply:
I don't have any strong opinion about it. I think that both work... but I'll not discard the need for the note to explain how to read the table data. I prefer to not let the user make assumptions.

@tremes WDYT?

Reply:
@jholecek-rh I think rule developers could easily understand it even not adding backported version here. It is better to add some explanation somewhere.

Reply:
I agree that user/reader should not guess and I think I agree with what Jan mentioned here. Perhaps we could provide the location with versions only in the case there have been any updates in the location.

Reply:
Some suggestions:

### Location in archive ( 👍 )
| Version   | Path											|
| --------- | --------------------------------------------------------	|
| any          | config/running_containers.json 					        |

### Location in archive ( 👎 )
| Version   | Path											|
| --------- | --------------------------------------------------------	|
| *               | config/running_containers.json 					        |

### Location in archive ( 😄 )
`config/running_containers.json`

@tremes @jholecek-rh @shzhou12 vote (by reacting) or suggest something else.

Reply:
I prefer the third one. It is simple and has no ambiguity when understanding the version range.

Reply:
Well, folks, I updated the information. Hopefully, it is better now.